lib/src/segy.c

```diff
@@ -597,32 +743,63 @@ int segy_close( segy_file* fp ) {
 }
 
 static int get_field( const char* header,
-                      const int* table,
-                      int field,
-                      int32_t* f ) {
+                      const uint8_t* table,
+                      field_data* fd) {
+    fd->datatype = table[ fd->field_id ];
```

Comment:
So field_id is already set before we get to this method and datatype is not?
Wondering if both shouldn't be set in one place.